packages/dd-trace/test/startup-log.spec.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const os = require('os')
+const tracerVersion = require('../lib/version')
+
+describe('startup logging', () => {
+  let semverVersion
+  let firstStderrCall
+  let secondStderrCall
+  before(() => {
+    sinon.stub(console, 'error')
+    semverVersion = require('semver/package.json').version
+    delete require.cache[require.resolve('../src/startup-log')]
+    const {
+      setStartupLogConfig,
+      setStartupLogPlugins,
+      setSamplingRules,
+      startupLog
+    } = require('../src/startup-log')
+    setStartupLogPlugins([{ name: 'http' }, { name: 'fs' }, { name: 'semver', versions: [] }])
+    setStartupLogConfig({
+      env: 'production',
+      enabled: true,
+      scope: 'async_hooks',
+      service: 'test',
+      hostname: 'example.com',
+      port: 4321,
+      debug: true,
+      analytics: true,
+      sampleRate: 1,
+      tags: { version: '1.2.3' },
+      logInjection: true,
+      runtimeMetrics: true
+
+    })
+    setSamplingRules(['rule1', 'rule2'])
+    startupLog({ message: 'Error: fake error' })
+    firstStderrCall = console.error.firstCall /* eslint-disable-line no-console */
```

Comment:
Why not just check if the underlying logger was called?

Reply:
I'm not sure it makes a difference here, and console.error is pretty easy to access.

Reply:
It's an implementation detail of the logger, so in general it would be better to not depend on it here. Not going to block the PR based on this if you prefer to keep it this way.